toolchain/check/testdata/interface/no_prelude/generic_method.carbon

```diff
@@ -0,0 +1,573 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// AUTOUPDATE
+// TIP: To test this file alone, run:
+// TIP:   bazel test //toolchain/testing:file_test --test_arg=--file_tests=toolchain/check/testdata/interface/no_prelude/generic_method.carbon
+// TIP: To dump output, run:
+// TIP:   bazel run //toolchain/testing:file_test -- --dump_output --file_tests=toolchain/check/testdata/interface/no_prelude/generic_method.carbon
+
+// --- generic_method.carbon
+
+library "[[@TEST_NAME]]";
+
+// T has index 0, Self has index 1, U has index 2.
+interface A(T:! type) {
+  fn F(U:! type, u: U) -> (T, Self, U);
+}
+
+class X {}
+class Y {}
+class Z {}
+
+impl Y as A(X) {
+  // Here, U has index 0. The specific used for A.F needs to renumber its U from
```

Comment:
Thanks for these comments :) I was trying to puzzle out if the index could be negative, and this test helped me out!